agent/src/agent/destination/anodot_api_client.py

```diff
@@ -58,13 +59,19 @@ def get_new_token(self):
         response.raise_for_status()
         return response.text.replace('"', '')
 
-    def _build_url(self, *args) -> str:
-        return urllib.parse.urljoin(self.url, '/'.join(['/api/v2', *args]))
+    def _build_url(self, *args, api_version='v2') -> str:
+        return urllib.parse.urljoin(self.url, '/'.join([f'/api/{api_version}', *args]))
 
     @endpoint
     def create_schema(self, schema):
         return self.session.post(self._build_url('stream-schemas'), json=schema, proxies=self.proxies)
 
+    @endpoint
+    def update_schema(self, schema):
+        url_ = self._build_url('stream-schemas', 'internal', f'?token={self.api_token}&id={schema["id"]}', api_version='v1')
+        self.session.headers.pop('Authorization')
+        return self.session.post(url_, json=schema, proxies=self.proxies)
```

Comment:
Option 1: do not use session, just do `requests.post(...)` 
Option 2: create 2 classes like `ApiV1Request` and `ApiV2Request`, each one of them will hold a separate session object. (Also build URL method can be moved there and there will be no need to pass the `api_version` flag)
In both cases `refresh_session_authorization()` should only be used for v2 requests. This can be achieved, for example, by making it a decorator also and calling it only for v2 functions. Or if you decide to go with v1/v2 classes, it can be managed there

Reply:
Decided to go after simpler Option 1, because `update_schema` - is one and only place where request to api/v1 is used